metagraph/src/graph/representation/canonical_dbg.cpp

```diff
@@ -195,21 +197,32 @@ void CanonicalDBG::append_next_rc_nodes(node_index node,
 
                 c = kmer::KmerExtractorBOSS::complement(c);
 
-                if (children[c] == npos)
+                if (children[c] != npos) {
+                    if (k_odd_) {
+                        common::logger->error(
+                            "Primary graph contains both forward and reverse complement: {} {} -> {} {}\t{} {}",
+                            node, graph_.get_node_sequence(node),
+                            children[c], graph_.get_node_sequence(children[c]),
+                            next, graph_.get_node_sequence(next));
```

Comment:
I think printing thousands of error messages is not helpful. Just throw an exception or make sure the behavior is correct, and hence, there is no need to print error messages. Otherwise, it's a half-correct function, isn't it?

Reply:
I think throwing an exception here is better since this message is only supposed to be printed if the input graph is invalid.

Reply:
Yes, in that case just print a one-line exception.